llvm-general-pure/src/LLVM/General/AST/Operand.hs

```diff
@@ -16,18 +16,24 @@ newtype MetadataNodeID = MetadataNodeID Word
 
 -- | <http://llvm.org/docs/LangRef.html#metadata>
 data MetadataNode 
-  = MetadataNode [Maybe Operand]
+  = MetadataNode [Maybe Metadata]
   | MetadataNodeReference MetadataNodeID
   deriving (Eq, Ord, Read, Show, Typeable, Data)
 
+data Metadata
+  = MDString String
+  | MDNode MetadataNode
+  | MDValue Operand
+  deriving (Eq, Ord, Read, Show, Typeable, Data)
+
 -- | An 'Operand' is roughly that which is an argument to an 'LLVM.General.AST.Instruction.Instruction'
 data Operand 
   -- | %foo
   = LocalReference Type Name
   -- | 'Constant's include 'LLVM.General.AST.Constant.GlobalReference', for \@foo
   | ConstantOperand Constant
-  | MetadataStringOperand String
-  | MetadataNodeOperand MetadataNode
+  -- MetadataStringOperand String
+  -- MetadataNodeOperand MetadataNode
```

Comment:
Why have you commented these out? The 3.8 language spec still says that "Metadata can be used as function arguments".